kubernetes-client-api/src/test/java/io/fabric8/kubernetes/client/internal/KubeConfigUtilsTest.java

```diff
@@ -115,6 +117,56 @@ void testGetUserAuthInfo() {
     assertEquals("test-token-2", authInfo.getToken());
   }
 
+  @Test
+  void getAuthInfo_when_authInfoExists_returnsAuthInfo() {
+    // given
+    Config config = getTestKubeConfig();
+    // when
+    NamedAuthInfo found = KubeConfigUtils.getAuthInfo(config, "test/api-test-com:443");
+    // then
+    assertThat(found).isNotNull();
+  }
+
+  @Test
+  void getAuthInfo_when_authInfoDoesntExist_returnsNull() {
+    // given
+    Config config = getTestKubeConfig();
+    // when
+    NamedAuthInfo found = KubeConfigUtils.getAuthInfo(config, "bogus");
+    // then
+    assertThat(found).isNull();
+  }
+
+  @Test
+  void hasAuthInfoNamed_when_authInfoExists_returnsTrue() {
+    // given
+    Config config = getTestKubeConfig();
+    // when
+    boolean hasIt = KubeConfigUtils.hasAuthInfoNamed(config, "test/api-test-com:443");
+    // then
+    assertThat(hasIt).isTrue();
+  }
+
+  @Test
+  void hasAuthInfoNamed_when_authInfoDoesntExist_returnsFalse() {
+    // given
+    Config config = getTestKubeConfig();
+    // when
+    boolean hasIt = KubeConfigUtils.hasAuthInfoNamed(config, "bogus");
+    // then
+    assertThat(hasIt).isFalse();
+  }
+
+  @Test
+  void hasAuthInfoNamed_when_hasNoAuthInfo_returnsFalse() {
+    // given
+    Config config = new ConfigBuilder().build();
+    // when
+    boolean hasIt = KubeConfigUtils.hasAuthInfoNamed(config, "test/api-test-com:443");
+    // then
+    assertThat(hasIt).isFalse();
+  }
```

Comment:
nit, I think these three tests can be made parameterized.

Reply:
@rohanKanojia not sure that we want this to be parametrized. The parametrized version that I just implemented is far less readable. See here:
```java
  @ParameterizedTest
  @MethodSource("hasAuthInfoNamed_arguments")
  void hasAuthInfoNamed(Config config, String authInfoName, Consumer<Boolean> consumer) {
    // given
    // when
    boolean hasIt = KubeConfigUtils.hasAuthInfoNamed(config, authInfoName);
    // then
    consumer.accept(hasIt);
  },

  static Stream<Arguments> hasAuthInfoNamed_arguments() {
    return Stream.of(
      // given config with authInfo, when getAuthInfoName with existing name, then should re
      Arguments.of(getTestKubeConfig(), "test/api-test-com:443", (Consumer<Boolean>) (hasIt -> assertThat(hasIt).isTrue())),
      Arguments.of(getTestKubeConfig(), "bogus", (Consumer<Boolean>) (hasIt -> assertThat(hasIt).isFalse())),
      Arguments.of(new ConfigBuilder().build(), "test/api-test-com:443", (Consumer<Boolean>) (hasIt -> assertThat(hasIt).isFalse()))
    );
  }
```